specutils/spectra/spectrum1d.py

```diff
@@ -662,19 +663,20 @@ def shift_spectrum_to(self, *, redshift=None, radial_velocity=None):
 
     @redshift.setter
     def redshift(self, val):
```

Comment:
Does this not work for you?
```suggestion
    @deprecated('1.8.0', alternative='set_redshift_to or shift_spectrum_to')
    def redshift(self, val):
```

Reply:
I meant to say in the description, I tried the decorators and they didn't work (threw an error on missing `__name__` attribute), but maybe that was because I put the `deprecated` decorator before the `setter` decorator...I can try this locally and see if it works in this order.

Reply:
Hmm maybe it just never works on a setter. It is okay to not use the decorator too. Pls lemme know if swapping works. 🤞

Reply:
Well, lo and behold, it does work with the decorators in the opposite order. Didn't think to try that until I saw your suggestion in that order.

Reply:
My order is purely accident. Thanks for trying! 😸